app/services/scheduler.py

```
@@ -402,6 +428,100 @@ def rotate_remote_code_job_func():
+                                ),
+                            )
+                        except Exception as e:
+                            logger.error(f"Failed to dispatch STORAGE_PERMISSION_ERROR notification: {e}")
+                else:
+                    if location.permissions_error is not None:
+                        logger.info(f"Permissions restored on cold storage '{location.name}'")
+                    location.permissions_error = None
+            except FileNotFoundError:
+                error = f"Path not found: {location.path}"
+                location.permissions_error = error
+                logger.warning(f"Cold storage path not found for '{location.name}': {location.path}")
```

Ruff (0.15.9)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)